.github/workflows/build.yml

```diff
@@ -0,0 +1,51 @@
+name: "Build"
+on:
+    push:
+      branches:
+        - main
+    pull_request:
+
+jobs:
+  build:
+    name: Build Job ${{ matrix.os }}
+    runs-on: ${{ matrix.os }}
+    env:
+      XCODE_VERSION: 12.5.1
+
+    strategy:
+      fail-fast: false
+      matrix:
+        os: [macos-latest]
+        java: ["11"]
+    
+    steps:
+      - name: Git checkout
+        uses: actions/checkout@v3
+      
+      - name: "Set up Java: ${{ matrix.java }}"
+        uses: actions/setup-java@v3
+        with:
+          java-version: ${{ matrix.java }}
+          distribution: "adopt"
+
+      - name: Cache Gradle packages
+        uses: actions/cache@v2
+        with:
+          path: |
+            ~/.gradle/caches
+            ~/.gradle/wrapper
+          key: ${{ runner.os }}-gradle-${{ hashFiles('**/*.gradle*', '**/gradle-wrapper.properties') }}
+          restore-keys: |
+            ${{ runner.os }}-gradle-
+
+      - name: Select Xcode 12
+        run: sudo xcode-select --switch /Applications/Xcode_${{ env.XCODE_VERSION }}.app && /usr/bin/xcodebuild -version
```

Comment:
Do we need the `XCODE_VERSION` from GH Action? Otherwise we can hardcode and put it directly in the make file

Reply:
I can put it directly into the make file